standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/messaging/json/JSONReloadMessage.java

```diff
@@ -133,6 +138,11 @@ public Partition apply(@Nullable Object input) {
             });
     }
 
+    @Override
+    public Partition getPtnObj() throws Exception {
```

Comment:
this method might return null somehow, will it break the compatibility? should we catch it on the impala side?

Reply:
I have added necessary gaurdrails on Impala side to avoid this null pointer exception. i.e., if getPtnObj() returns null then we should check getPartitionObjs() method.

Reply:
I have a little confusion here, if the getPtnObj() is null, then we would reload the whole table, would this ignore the partitions(the `ptns`) in this reload event?

Reply:
@saihemanth-cloudera should we consider this https://github.com/apache/hive/pull/6058#discussion_r2370964841, and remove the always null parameter?